.github/workflows/dev.yaml

```diff
@@ -0,0 +1,27 @@
+name: Dev CI
+
+on:
+  push:
+    branches: [ dev ]
```

Comment:
question: is it different from `release.yml` ? Why couldn't we put the `dev` branch there?

Reply:
To have better control 😅 I wanted this branch to only publish after running the tests. But yes, we can just add the `dev` branch to the `scala.yaml` and `release.yaml` and we should be good for what we want to do 😅

Reply:
Simplified @ [1e2bf75](https://github.com/SwissBorg/akka-persistence-postgres/pull/352/commits/1e2bf75e43853c662cf4f80b5f2d68c0386e5bb9)